### ISSUE TITLE
Does support for avr32-gcc works?

### ISSUE THREAD

**Issue by cozycactus** [clangd/clangd#1894]
i have this in log :
System include extraction: not allowed driver /home/cozy/Downloads/as4e-ide/plugins/com.atmel.avr.toolchains.linux.x86_64_3.0.0.201009140852/os/linux/x86_64/bin/avr32-gcc

**Comment by i-ky** [comment]
"not allowed driver" does not mean anything bad per se, it just means that clangd is not allowed to launch compiler binary to extract system include paths from there. Read about [`--query-driver` option](https://clangd.llvm.org/guides/system-headers#query-driver) for more information on that. Note that clangd has a number of heuristics to guess include paths without running compiler binary. But how well they work may vary.

**Comment by HighCommander4** [comment]
Right, usually in a case like this you will want to add `--query-driver=/home/cozy/Downloads/as4e-ide/plugins/com.atmel.avr.toolchains.linux.x86_64_3.0.0.201009140852/os/linux/x86_64/bin/avr32-gcc` to clangd's arguments.

**Comment by cozycactus** [comment]
@HighCommander4 @i-ky 
i already had this in arguments :
I[11:27:25.197] clangd version 17.0.3 (https://github.com/llvm/llvm-project 888437e1b60011b8a375dd30928ec925b448da57)
I[11:27:25.197] Features: linux+grpc
I[11:27:25.197] PID: 3830
I[11:27:25.197] Working directory: /home/cozy/sdr-widget
I[11:27:25.197] argv[0]: /home/cozy/.config/Code/User/globalStorage/llvm-vs-code-extensions.vscode-clangd/install/17.0.3/clangd_17.0.3/bin/clangd
I[11:27:25.197] argv[1]: -query-driver=/home/cozy/Downloads/as4e-ide/plugins/com.atmel.avr.toolchains.linux.x86_64_3.0.0.201009140852/os/linux/x86_64/bin/avr32-gcc
I[11:27:25.197] argv[2]: -background-index
I[11:27:25.197] argv[3]: -log
I[11:27:25.197] argv[4]: verbose
I[11:27:25.197] argv[5]: -compile-commands-dir
I[11:27:25.197] argv[6]: .
V[11:27:25.198] User config file is /home/cozy/.config/clangd/config.yaml
I[11:27:25.198] Starting LSP over stdin/stdout

**Comment by i-ky** [comment]
Could you please share the full log?

**Comment by HighCommander4** [comment]
> @i-ky https://logpaste.com/Of3j0b9O

This log doesn't contain any "System include extraction: not allowed driver" error, so it sounds like you've resolved the problem in the issue description?

**Comment by cozycactus** [comment]
> > @i-ky https://logpaste.com/Of3j0b9O
> 
> This log doesn't contain any "System include extraction: not allowed driver" error, so it sounds like you've resolved the problem in the issue description?

yes, this specific issue

**Comment by i-ky** [comment]
> @i-ky https://logpaste.com/Of3j0b9O

Hm, `--quote-driver` and system include extraction seems to work fine, however:
```
E[11:33:50.590] System include extraction: invalid target "avr32", ignoring
```
... and then:
```
I[11:33:50.590] System includes extractor: successfully executed /home/cozy/Downloads/as4e-ide/plugins/com.atmel.avr.toolchains.linux.x86_64_3.0.0.201009140852/os/linux/x86_64/bin/avr32-gcc
	got includes: "/home/cozy/Downloads/as4e-ide/plugins/com.atmel.avr.toolchains.linux.x86_64_3.0.0.201009140852/os/linux/x86_64/bin/../lib/gcc/avr32/4.3.3/include, /home/cozy/Downloads/as4e-ide/plugins/com.atmel.avr.toolchains.linux.x86_64_3.0.0.201009140852/os/linux/x86_64/bin/../lib/gcc/avr32/4.3.3/include-fixed, /home/cozy/Downloads/as4e-ide/plugins/com.atmel.avr.toolchains.linux.x86_64_3.0.0.201009140852/os/linux/x86_64/bin/../lib/gcc/avr32/4.3.3/../../../../avr32/include"
	got target: ""
```
(note the empty target)

Perhaps, clang does not support this target (or it is named differently in clang vs. GCC) and consequently does not set necessary target-specific defines, which lead to type definition being inactive (under some kind of `#ifdef`).

**Comment by HighCommander4** [comment]
Since clangd is built on top of clang, a good rule of thumb is, if you can compile your code with clang as the compiler, you should be able to get clangd to work for it as well.

I'm not familiar the avr32 target and I don't know if clang supports it. You may have better luck finding an answer to that on the [LLVM Discourse](https://discourse.llvm.org/), or in an avr32-specific forum.
